cleanrl/rnd_ppo.py

```diff
@@ -0,0 +1,1079 @@
+        if args.gae:
+            ext_advantages = torch.zeros_like(rewards).to(device)
+            int_advantages = torch.zeros_like(curiosity_rewards).to(device)
+            ext_lastgaelam = 0
+            int_lastgaelam = 0
+            for t in reversed(range(args.num_steps)):
+                if t == args.num_steps - 1:
+                    ext_nextnonterminal = 1.0 - next_done
+                    int_nextnonterminal = 1.0
+                    ext_nextvalues = last_value_ext
+                    int_nextvalues = last_value_int
+                else:
+                    ext_nextnonterminal = 1.0 - dones[t+1]
+                    int_nextnonterminal = 1.0
+                    ext_nextvalues = ext_values[t+1]
+                    int_nextvalues = int_values[t+1]
+                ext_delta = rewards[t] + args.gamma * ext_nextvalues * ext_nextnonterminal - ext_values[t]
+                int_delta = curiosity_rewards[t] + args.int_gamma * int_nextvalues * int_nextnonterminal - int_values[t]
+                ext_advantages[t] = ext_lastgaelam = ext_delta + args.gamma * args.gae_lambda * ext_nextnonterminal * ext_lastgaelam
+                int_advantages[t] = int_lastgaelam = int_delta + args.int_gamma * args.gae_lambda * int_nextnonterminal * int_lastgaelam
+            ext_returns = ext_advantages + ext_values
+            int_returns = int_advantages + int_values
+        else:
+            ext_returns = torch.zeros_like(rewards).to(device)
+            int_returns = torch.zeros_like(curiosity_rewards).to(device)
+            for t in reversed(range(args.num_steps)):
+                if t == args.num_steps - 1:
+                    ext_nextnonterminal = 1.0 - next_done
+                    int_nextnonterminal = 1.0
+                    ext_next_return = last_value_ext
+                    int_next_return = last_value_int
+                else:
+                    ext_nextnonterminal = 1.0 - dones[t+1]
+                    int_nextnonterminal = 1.0
+                    ext_next_return = ext_returns[t+1]
+                    int_next_return = int_returns[t+1]
+                ext_returns[t] = rewards[t] + args.gamma * ext_nextnonterminal * ext_next_return
+                int_returns[t] = curiosity_rewards[t] + args.int_gamma * int_nextnonterminal * int_next_return
+            ext_advantages = ext_returns - ext_values
+            int_advantages = int_returns - int_values
+
+    # flatten the batch
+    b_obs = obs.reshape((-1,)+envs.observation_space.shape)
+    b_logprobs = logprobs.reshape(-1)
+    b_actions = actions.reshape((-1))
+    b_ext_advantages = ext_advantages.reshape(-1)
+    b_int_advantages = int_advantages.reshape(-1)
+    b_ext_returns = ext_returns.reshape(-1)
+    b_int_returns = int_returns.reshape(-1)
+    b_ext_values = ext_values.reshape(-1)
+
+    b_advantages = b_int_advantages * args.int_coef + b_ext_advantages * args.ext_coef
+    del b_int_advantages, b_ext_advantages
```

Comment:
I assume this will be taken care for by the garbage collector. I suggest removing it.